django_elasticsearch_dsl/documents.py

```diff
@@ -156,11 +156,19 @@ def parallel_bulk(self, actions, **kwargs):
         # the result is currently not used upstream anyway.
         return (1, [])
 
+    def document_id(self, object_instance):
```

Comment:
Can you change the method name to `get_document_id`. `document_id` seems like a `property` from developer point of view.

Reply:
Can you also add a test for this?

Reply:
I get your point concerning the method name, but I have a feeling that `get_document_id` may be misleading, as this is not really a getter. It's more like a setter. It is basically the same as the `prepare_<field name>` methods, but for the document id instead of one of its fields.

I did not use the name `prepare_document_id` or `prepare_id`, as this would conflict with the methods that would set the `document_id` or `id` field inside the document.

Would you be ok with one of the following names?
- `set_id`
- `set_document_id`
- `determine_document_id`
- `determine_id`

Tell me which one you prefer and I'll do the changes.

And I'll add tests as well.

Reply:
I think it is not setter as it is not setting any atribute to the object itself. the method is returning the id so from independent prespective, it is a getter. You can change it to something like `generate_id`, which maybe more explicit.

Reply:
Yes, you're right, it's not a setter either. Ok for `generate_id` then.